pkg_resources/__init__.py

```diff
@@ -2477,11 +2477,26 @@ def _dep_map(self):
                             extra, marker = extra.split(':', 1)
                             if invalid_marker(marker):
                                 # XXX warn
-                                reqs=[]
+                                reqs = []
                             elif not evaluate_marker(marker):
-                                reqs=[]
+                                reqs = []
                         extra = safe_extra(extra) or None
-                    dm.setdefault(extra,[]).extend(parse_requirements(reqs))
+                    try:
+                        dm.setdefault(extra, []).extend(
+                            parse_requirements(reqs))
+                    except (
+                        packaging.requirements.InvalidRequirement,
```

Comment:
Here InvalidRequirement gets translated to a RequirementParseError - that could have implications later if some part of the system expects to discriminate between the two exceptions.